pkg/utils/deprecation/auto_mode.go

```diff
@@ -0,0 +1,69 @@
+package deprecation
+
+import (
+	"github.com/kris-nova/logger"
+	"github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+)
+
+// AutoModeDeprecationConfig holds deprecation warning settings
+type AutoModeDeprecationConfig struct {
+	Message          string
+	DocumentationURL string
+	LogLevel         string
+}
+
+// DefaultAutoModeDeprecationConfig returns default deprecation warning config
+func DefaultAutoModeDeprecationConfig() *AutoModeDeprecationConfig {
+	return &AutoModeDeprecationConfig{
+		Message: "Auto Mode will be enabled by default in an upcoming release of eksctl. " +
+			"This means managed node groups and managed networking add-ons will no longer be created by default. " +
+			"To maintain current behavior, explicitly set 'autoModeConfig.enabled: false' in your cluster configuration.",
+		DocumentationURL: "https://eksctl.io/usage/auto-mode/",
+		LogLevel:         "warning",
+	}
+}
+
+// CheckAutoModeDeprecation shows deprecation warning if needed
+func CheckAutoModeDeprecation(cfg *v1alpha5.ClusterConfig) {
+	if shouldShowAutoModeDeprecationWarning(cfg) {
+		logAutoModeDeprecationWarning(DefaultAutoModeDeprecationConfig())
+	}
+}
+
+// CheckAutoModeDeprecationWithConfig shows deprecation warning with custom config
+func CheckAutoModeDeprecationWithConfig(cfg *v1alpha5.ClusterConfig, deprecationConfig *AutoModeDeprecationConfig) {
+	if shouldShowAutoModeDeprecationWarning(cfg) {
+		logAutoModeDeprecationWarning(deprecationConfig)
+	}
+}
+
+// shouldShowAutoModeDeprecationWarning returns true if warning should be shown
+func shouldShowAutoModeDeprecationWarning(cfg *v1alpha5.ClusterConfig) bool {
+	// Show warning only when AutoMode is nil or Enabled is nil (not explicitly set)
+	if cfg.AutoModeConfig == nil {
+		return true
+	}
+	if cfg.AutoModeConfig.Enabled == nil {
+		return true
+	}
+	// Don't show warning when explicitly set to true or false
+	return false
```

Comment:
nit: isn't this just `return cfg.AutoModeConfig.Enabled == nil` ?

You could even combine it with the above and this whole function could be a 1-liner `return cfg.AutoModeConfig == nil || cfg.AutoModeConfig.Enabled == nil` if you wanted